packages/@lwc/ssr-compiler/src/compile-template/shared.ts

```diff
@@ -129,7 +129,12 @@ export function getChildAttrsOrProps(
 
                 return b.property('init', key, b.literal(type === 'Attribute' ? '' : value.value));
             } else if (value.type === 'Identifier' || value.type === 'MemberExpression') {
-                const propValue = expressionIrToEs(value, cxt);
+                let propValue = expressionIrToEs(value, cxt);
+
+                if (name === 'class') {
+                    propValue = b.callExpression(b.identifier('normalizeClass'), [propValue]);
+                }
```

Comment:
We are missing a `cxt.import('normalizeClass')` here and not detecting it because the element path (`<button>`) and component path (`<x-child>`) are being shared in the same parent component. So one usage of `normalizeClass` is piggybacking off of the other's `import`. I'd recommend splitting the fixture into two separate ones which would have surfaced the bug.

Reply:
Ah nice catch!